app/models/subnet.rb

```diff
@@ -1,17 +1,37 @@
 require 'ipaddr'
+
 class Subnet < ActiveRecord::Base
+  def self.default_sti_class
+    'Subnet::Ipv4' if self.name == 'Subnet'
+  end
+
+  IP_FIELDS = [:network, :mask, :gateway, :dns_primary, :dns_secondary, :from, :to]
+  REQUIRED_IP_FIELDS = [:network, :mask]
+  SUBNET_TYPES = {:'Subnet::Ipv4' => N_('IPv4')}
   BOOT_MODES = {:static => N_('Static'), :dhcp => N_('DHCP')}
   IPAM_MODES = {:dhcp => N_('DHCP'), :db => N_('Internal DB'), :none => N_('None')}
 
   include Authorizable
+  include Foreman::STI
   extend FriendlyId
   friendly_id :name
   include Taxonomix
   include Parameterizable::ByIdName
   include EncOutput
-  attr_accessible :name, :network, :mask, :gateway, :dns_primary, :dns_secondary, :ipam, :from,
+  attr_accessible :name, :type, :network, :mask, :gateway, :dns_primary, :dns_secondary, :ipam, :from,
     :to, :vlanid, :boot_mode, :dhcp_id, :dhcp, :tftp_id, :tftp, :dns_id, :dns, :domain_ids, :domain_names,
-    :subnet_parameters_attributes
+    :subnet_parameters_attributes, :cidr
+
+  def self.inherited(child)
+    child.instance_eval do
```

Comment:
This sets the modal name for the STI subclasses to "Subnet". It would be e.g. 'subnet_ipv4' with out this and break paths, forms, etc.
Does an inline comment for this makes sense?